deviations/views.py

```diff
@@ -16,18 +20,77 @@ class ListDeadlinesView(CourseInstanceBaseView):
     access_mode = ACCESS.TEACHER
     template_name = "deviations/list_dl.html"
 
-    def get_common_objects(self):
+    def get_common_objects(self) -> None:
         super().get_common_objects()
-        self.deviations = (
+
+        self.deviation_groups = self.get_deviation_groups()
+        self.note("deviation_groups")
+
+    def get_deviation_groups(self) -> Iterable[Tuple[List[DeadlineRuleDeviation], bool, Optional[str]]]:
```

Comment:
I spent a lot of time trying different methods for grouping the deviations by user and module (template tags, query trickery, etc), and the current method that uses `itertools.groupby` is definitely both the fastest (it barely has any performance effect with 2000 deviations) and cleanest way in my opinion.